code/__DEFINES/maths.dm

```diff
@@ -239,3 +239,35 @@
 /// Gives the number of pixels in an orthogonal line of tiles.
 #define TILES_TO_PIXELS(tiles)			(tiles * PIXELS)
 // )
+
+// value placeholders for ADDCLAMP macro
+GLOBAL_VAR(addclamp_respect)
+GLOBAL_VAR(addclamp_add)
+GLOBAL_VAR(addclamp_low)
+GLOBAL_VAR(addclamp_high)
+
+/// Similar to clamp(). This is identical to 'x + y' thing, but add to 'x' will be limited within low-high range.
+/// If val_respected is already outside of low-high range, it will return val_respected, instead of the value within the range.
+/// Note: First wrapped codes are value placeholders because this define can be expensive when parameter takes a proc
+#define ADDCLAMP(val_respected, val_to_add, low, high) ( \
+	( \
+		((GLOB.addclamp_respect = (val_respected))||1) && \
+		((GLOB.addclamp_add = (val_to_add))||1) && \
+		((GLOB.addclamp_low = (low))||1) && \
+		((GLOB.addclamp_high = (high))||1) \
+	) && \
+	(GLOB.addclamp_respect + GLOB.addclamp_add) >= GLOB.addclamp_high \
+	? max(GLOB.addclamp_respect, GLOB.addclamp_high) \
+	: (GLOB.addclamp_respect + GLOB.addclamp_add) <= GLOB.addclamp_low \
+	? min(GLOB.addclamp_respect, GLOB.addclamp_low) \
+	: GLOB.addclamp_respect + GLOB.addclamp_add \
+)
```

Comment:
```
(val + add) > max ? (val + add) : val < min ? val : clamp(val + add, min, max)
```

val 15, add 5, low 0, high 15 (15 + 5 > 15, so returns 15).
val 5, add 10, low 0, high 10 (5 + 10 = 15, so returns 10)
val 5, add -10, low 0, high 10 (5 - 10 = -5, so returns 0)
val -5, add -10, low 0, high 10 (-5 - 15 = -15, so returns -5)

This passes the basic tests although I haven't fully understood the point of the edge cases so needs double checking

Reply:
Wait why even make these global variables, they are litreally just assigned to values?